conjure_oxide/tests/integration/basic/comprehension/01/input-expected-rule-trace-human.txt

```diff
@@ -0,0 +1,207 @@
+Model before rewriting:
+
+find m: matrix indexed by [[int(1..5)]] of int(1..5)
+find x: int(1..2)
+find y: int(1..2)
+
+such that
+
+(x = 1),
+and([((y < 5)) -> ((m[x] = x)) | x: int(1..5),(x < 4)])
+
+--
+
+(y < 5), 
+   ~~> lt_to_leq ([("Minion", 8400)]) 
+(y <= Sum([5, -1])) 
+
+--
+
+Sum([5, -1]), 
+   ~~> apply_eval_constant ([("Constant", 9001)]) 
+4 
+
+--
+
+(x < 4), 
+   ~~> lt_to_leq ([("Minion", 8400)]) 
+(x <= Sum([4, -1])) 
+
+--
+
+Sum([4, -1]), 
+   ~~> apply_eval_constant ([("Constant", 9001)]) 
+3 
+
+--
+
+m, 
+   ~~> select_representation ([("Base", 8000)]) 
+m#matrix_to_atom 
+new variables:
+  find m#matrix_to_atom_1: int(1..5)
+  find m#matrix_to_atom_2: int(1..5)
+  find m#matrix_to_atom_3: int(1..5)
+  find m#matrix_to_atom_4: int(1..5)
+  find m#matrix_to_atom_5: int(1..5)
+--
+
+m#matrix_to_atom[x], 
+   ~~> index_to_bubble ([("Bubble", 6000)]) 
+{m#matrix_to_atom[x] @ and([__inDomain(x,int(1..5));int(1..)])} 
+
+--
+
+({m#matrix_to_atom[x] @ and([__inDomain(x,int(1..5));int(1..)])} = x), 
+   ~~> bubble_up ([("Bubble", 8900)]) 
+{(m#matrix_to_atom[x] = x) @ and([and([__inDomain(x,int(1..5));int(1..)]);int(1..)])} 
+
+--
+
+{(m#matrix_to_atom[x] = x) @ and([and([__inDomain(x,int(1..5));int(1..)]);int(1..)])}, 
+   ~~> expand_bubble ([("Bubble", 8900)]) 
+and([(m#matrix_to_atom[x] = x),and([and([__inDomain(x,int(1..5));int(1..)]);int(1..)]);int(1..)]) 
+
+--
+
+and([and([__inDomain(x,int(1..5));int(1..)]);int(1..)]), 
+   ~~> remove_unit_vector_and ([("Base", 8800)]) 
+and([__inDomain(x,int(1..5));int(1..)]) 
+
+--
+
+and([__inDomain(x,int(1..5));int(1..)]), 
+   ~~> remove_unit_vector_and ([("Base", 8800)]) 
+__inDomain(x,int(1..5)) 
+
+--
+
+((y <= 4)) -> (and([(m#matrix_to_atom[x] = x),__inDomain(x,int(1..5));int(1..)])), 
```

Comment:
this happens before comprehension unrolling, I suspect because it doesn't refer to the quantified variable? can you add another test case to show what happens when it refers to the quantified variable?

Reply:
> can you add another test case to show what happens when it refers to the quantified variable?
I'm not sure I get what you mean here: could you give an example?

Reply:
Something like:

```
find  m,n: matrix indexed by [int(1..5)] of int(1..5) 
such that and([n[i] = 6-i | i: int(1..5) ]) $ fix n so we don't have too many solutions
such that and([m[i] = i | i: int(1..5), n[i] < 5,i<4])
```

notice I am using n[i] instead of y